src/strategies/shinobi-staked-in-tge-farms/index.ts

```diff
@@ -0,0 +1,158 @@
+import { formatUnits } from '@ethersproject/units';
+import { multicall } from '../../utils';
+import { subgraphRequest } from '../../utils';
+
+export const author = 'scrappydew';
+export const version = '0.1.0';
+
+const SHINOBI_SUBGRAPH_URL =
+  'https://graphnode.octano.dev/subgraphs/name/octanolabs/shinobi';
+
+const getAllLP = async (skip, stakedLP, snapshot) => {
+  const params = {
+    pairs: {
+      __args: {
+        where: {
+          id_in: stakedLP.map((address) => address.toLowerCase())
+        },
+        skip: skip,
+        first: 1000
+      },
+      id: true,
+      reserve0: true,
+      reserve1: true,
+      totalSupply: true,
+      token0: {
+        id: true,
+        symbol: true
+      },
+      token1: {
+        id: true,
+        symbol: true
+      }
+    }
+  };
+  if (snapshot !== 'latest') {
+    // @ts-ignore
+    params.pairs.__args.block = { number: snapshot };
+  }
+  try {
+    const response = await subgraphRequest(SHINOBI_SUBGRAPH_URL, params);
+    return response.pairs;
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const getLP = async (stakedLP, snapshot) => {
+  let lp = [];
+  let results = [];
+  do {
+    results = await getAllLP(lp.length, stakedLP, snapshot);
+    lp = lp.concat(results);
+  } while (results.length === 1000);
```

Comment:
How many times subgraph will be called here ?  There is a limit of 5 API calls for a strategy https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy

Reply:
i copied this code from an existing strategy that uses that same code, I didnt modify it. 

i can look for another example